packages/core/saved-objects/core-saved-objects-base-server-internal/src/validation/validator.test.ts

```diff
@@ -127,39 +129,71 @@ describe('Saved Objects type validator', () => {
 
       data = createMockObject({ typeMigrationVersion: '3.5.0' });
       validator.validate(data, '4.5.0');
-      expect(getCalledVersion()).toEqual('4.3.0');
+      expect(getCalledVersion()).toEqual('3.0.0');
     });
 
     it('should use the correct schema for documents with typeMigrationVersion', () => {
-      let data = createMockObject({ typeMigrationVersion: '3.2.0' });
+      const data = createMockObject({ typeMigrationVersion: '3.2.0' });
+      validator.validate(data);
+      expect(getCalledVersion()).toEqual('3.0.0');
+    });
+
+    it('should use the correct schema for documents with typeMigrationVersion higher than default when not a valid virtual model version', () => {
+      const data = createMockObject({ typeMigrationVersion: '3.5.0' });
+      validator.validate(data);
+      expect(getCalledVersion()).toEqual('3.0.0');
+    });
+
+    it('should fall back to most recent schema for documents with migrationVersion', () => {
+      let data = createMockObject({
+        migrationVersion: {
+          [type]: '4.6.0',
+        },
+      });
       validator.validate(data);
       expect(getCalledVersion()).toEqual('3.0.0');
 
       jest.clearAllMocks();
 
-      data = createMockObject({ typeMigrationVersion: '3.5.0' });
+      data = createMockObject({
+        migrationVersion: {
+          [type]: '3.0.0',
+        },
+      });
       validator.validate(data);
-      expect(getCalledVersion()).toEqual('3.5.0');
+      expect(getCalledVersion()).toEqual('3.0.0');
     });
 
-    it('should use the correct schema for documents with migrationVersion', () => {
+    it('should use the correct schema for documents with migrationVersion higher than default when not a valid virtual model version', () => {
       let data = createMockObject({
         migrationVersion: {
           [type]: '4.6.0',
         },
       });
       validator.validate(data);
-      expect(getCalledVersion()).toEqual('4.3.0');
+      expect(getCalledVersion()).toEqual('3.0.0');
 
       jest.clearAllMocks();
 
       data = createMockObject({
         migrationVersion: {
-          [type]: '4.0.0',
+          [type]: '3.0.0',
         },
       });
       validator.validate(data);
-      expect(getCalledVersion()).toEqual('4.0.0');
+      expect(getCalledVersion()).toEqual('3.0.0');
+    });
+
+    it('should use the correct schema for documents with virtualModelVersion', () => {
+      const data = createMockObject({ typeMigrationVersion: '10.1.0' });
+      validator.validate(data);
+      expect(getCalledVersion()).toEqual('10.1.0');
+    });
```

Comment:
NIT: Instead of registering a `10.1.0` schema, I would register one of `10.2.0`  and then assert that

- a document with version `10.1.0` uses schema  `4.3.0`
- a document with version `10.2.0` uses schema  `10.2.0`
- a document with version `10.3.0` uses schema  `10.2.0`

Reply:
> a document with version `10.1.0` uses schema `4.3.0`

The default version is `3.3.0` in the tests, so the schema that will be used is `3.0.0`